### PR TITLE
fix: Get and delete event favourite using event_id

### LINKED ISSUE #5857: Delete User Favorite By EventId
**Is your feature request related to a problem? Please describe.**
Current Implementation of user-favorite-events allows deletion of events by using the id given by the server.
**Describe the solution you'd like**
There should be an option to delete the event using eventId rather the user-favorite-event id

### PR DESCRIPTION
**Pull request by prateekj117**
Fixes #5857 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [X] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

- [X] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
Delete a favourite event using Event ID.

### CODE REVIEW

**Comment by prateekj117** [comment]
@uds5501 @mrsaicharan1 @shreyanshdwivedi Can you guys please have a look at this. I have tested everything locally, generated the docs, checked the response, everything works fine. Can you guys please see why the tests are failing.

**Comment by mrsaicharan1** [comment]
Mostly the docs which have been updated in the `.apib` fails. Just the check the request once more maybe?

**Comment by prateekj117** [comment]
@mrsaicharan1 Yaa, dredd is failing, that's sure. I would try one more thing.

File: docs/api/api_blueprint.apib [hunk 1]
```diff
@@ -25594,9 +25594,9 @@ This Group's APIs can be used for adding a particular event to the favourite lis
             }
         }
 
-## Favourite Events Detail [/v1/user-favourite-events/{user_favourite_event_id}]
+## Favourite Events Detail [/v1/user-favourite-events/{event_id}]
 + Parameters
-    + user_favourite_event_id: 1 (integer) - ID of the Favourite Event
+    + event_id: 1 (integer) - ID of the Event in the form of an integer
```
Comment (iamareebjamal) [inline docs/api/api_blueprint.apib]: Changing name of parameters won't fix anything.

Comment (prateekj117) [reply docs/api/api_blueprint.apib]: @iamareebjamal But it should be changed, right?

Comment (iamareebjamal) [reply docs/api/api_blueprint.apib]: Yes, your version is more explicit, but if you changed it in hopes that it'll fix the error, then it won't

**Comment by prateekj117** [comment]
@uds5501 @mrsaicharan1 @shreyanshdwivedi While using current_user, it shows `current_user` is None. Any idea why?

**Comment by shreyanshdwivedi** [comment]
@prateekj117 are you trying FE or Postman? If postman maybe you haven't authenticated yourself

**Comment by prateekj117** [comment]
@shreyanshdwivedi Trying postman. I have authenticated myself.

File: app/api/user_favourite_events.py [hunk 2]
```diff
@@ -79,11 +81,52 @@ class UserFavouriteEventDetail(ResourceDetail):
     """
     User Favourite Events detail by id
     """
+    def before_get_object(self, view_kwargs):
+        if 'Authorization' in request.headers:
+            _jwt_required(app.config['JWT_DEFAULT_REALM'])
+        else:
+            raise ForbiddenException({'source': ''}, 'Only Authorized Users can view an event')
```
Comment (iamareebjamal) [inline app/api/user_favourite_events.py]: Use decorator

File: app/api/user_favourite_events.py [hunk 3]
```diff
@@ -79,11 +81,52 @@ class UserFavouriteEventDetail(ResourceDetail):
...
+                    if user_favourite_event.event_id is not None:
+                        view_kwargs['id'] = user_favourite_event.id
+                else:
+                    view_kwargs['id'] = None
+
+    def before_delete_object(self, view_kwargs):
+        if 'Authorization' in request.headers:
+            _jwt_required(app.config['JWT_DEFAULT_REALM'])
+        else:
+            raise ForbiddenException({'source': ''}, 'Only Authorized Users can delete an event')
```
Comment (iamareebjamal) [inline app/api/user_favourite_events.py]: Use decorator

File: app/api/user_favourite_events.py [hunk 4]
```diff
@@ -79,11 +81,52 @@ class UserFavouriteEventDetail(ResourceDetail):
...
+
+        if view_kwargs.get('id') is not None:
+            try:
+                user_favourite_event = UserFavouriteEvent.query.filter_by(
+                    user=current_user, event_id=view_kwargs['id']).first()
+            except NoResultFound:
+                raise ObjectNotFound({'source': ''}, "Object: not found")
+            else:
+                if user_favourite_event:
+                    if user_favourite_event.event_id is not None:
```
Comment (iamareebjamal) [inline app/api/user_favourite_events.py]: Why this check?

File: app/api/user_favourite_events.py (hunk 2 above)
Comment (prateekj117) [reply app/api/user_favourite_events.py]: @iamareebjamal As discussed earlier, decorators won't work here.

Comment (iamareebjamal) [reply app/api/user_favourite_events.py]: Why?

Comment (prateekj117) [reply app/api/user_favourite_events.py]: @iamareebjamal decorators cannot be declared in function, secondly, we can use has_access, which again requires `current_user`, which is returning null.

File: app/api/user_favourite_events.py (hunk 4 above)
Comment (prateekj117) [reply app/api/user_favourite_events.py]: @iamareebjamal Not required, I modified the conditional, please review.

File: app/api/user_favourite_events.py [hunk 5]
```diff
@@ -79,11 +81,45 @@ class UserFavouriteEventDetail(ResourceDetail):
     """
     User Favourite Events detail by id
     """
+    @jwt_required()
```
Comment (iamareebjamal) [inline app/api/user_favourite_events.py]: paranthesis is redundant

File: app/api/user_favourite_events.py [hunk 6]
```diff
@@ -79,11 +81,45 @@ class UserFavouriteEventDetail(ResourceDetail):
...
+            except NoResultFound:
+                raise ObjectNotFound({'source': ''}, "Object: not found")
+            else:
+                if user_favourite_event is not None:
+                    view_kwargs['id'] = user_favourite_event.id
+                else:
+                    view_kwargs['id'] = None
+
+    @jwt_required()
+    def before_delete_object(self, view_kwargs):
```
Comment (iamareebjamal) [inline app/api/user_favourite_events.py]: This is unneeded. Remove

File: app/api/user_favourite_events.py [hunk 7]
```diff
@@ -79,11 +81,45 @@ class UserFavouriteEventDetail(ResourceDetail):
...
     """
+    @jwt_required()
+    def before_get_object(self, view_kwargs):
+
+        if view_kwargs.get('id') is not None:
+            try:
+                user_favourite_event = UserFavouriteEvent.query.filter_by(
+                    user=current_user, event_id=view_kwargs['id']).first()
+            except NoResultFound:
+                raise ObjectNotFound({'source': ''}, "Object: not found")
```
Comment (iamareebjamal) [inline app/api/user_favourite_events.py]: Set proper source

File: app/api/user_favourite_events.py (hunk 5 above)
Comment (prateekj117) [reply app/api/user_favourite_events.py]: @iamareebjamal It shows error when not using paranthesis.
Screenshot from 2019-06-22 14-20-29

**Comment by prateekj117** [comment]
@iamareebjamal Please review.

File: app/api/user_favourite_events.py (hunk 5 above)
Comment (iamareebjamal) [reply app/api/user_favourite_events.py]: OK

**Comment by iamareebjamal** [comment]
Follow semantic PR title

**Comment by prateekj117** [comment]
@iamareebjamal Please review.

**Comment by iamareebjamal** [comment]
I thought this was already merged

**Pull request merged**